### PR TITLE
Feature/hii 17 box

### PR DESCRIPTION
**Pull request by n3op2**
## What

an AppBar component for main navigation which can contain children, ideally menu items, but can be anything, takes css properties as props, more in the screenshots below and this will be updated

## TODO
- [x] - component testing
- [x] - convert into ul/li as it meant to be
- [x] - update with menu items, when menu item has been created
- [x] - a better playground e.g. allows nicer iteraction with props and more props

## Screenshoots
Screenshot 2022-12-13 at 19 50 02
Screenshot 2022-12-13 at 20 22 13
Screenshot 2022-12-13 at 20 22 08

### CODE REVIEW

File: src/AppBar/AppBar.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import React from 'react'
+import { Story, Meta } from '@storybook/react'
+
+import AppBar from './index'
+
+export default {
```
Comment (mattdean-digicatapult) [inline src/AppBar/AppBar.stories.tsx:6]: nice, I did not have a good solution to this on my PR. Will update to match

File: src/AppBar/AppBar.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  children,
+  search = false,
+  shadow = false,
+  theme = Theme,
+  ...props
+}) => {
+  return (
+    <Wrapper {...props} theme={theme} shadow={shadow}>
+      <Bar>{children}</Bar>
+      {search && <Search {...props} />}
```
Comment (mattdean-digicatapult) [inline src/AppBar/AppBar.tsx:89]: should search not be part of children?

File: src/AppBar/AppBar.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+`
+
+const ItemStyle = styled('div')<AppBarProps>`
+  font-family: 'Roboto';
+  font-style: normal;
+  font-weight: 700;
+  padding: 0px 10px;
+  text-transform: uppercase;
+  font-size: 16px;
+  line-height: 15px;
```
Comment (mattdean-digicatapult) [inline src/AppBar/AppBar.tsx:69]: Not so keen on the `px` sizing here. For padding and min/max height it's probably bad but font sizing should certainly be in em's

File: src/AppBar/AppBar.tsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  Item?: React.FC<PropsWithChildren<AppBarProps>>
+}
+
+const Theme = {
+  primary: '#FFA',
+  accent: '#000',
+}
+
+// TODO props per dom element as well as all for appbar
+const Wrapper = styled('div')<AppBarProps>`
```
Comment (mattdean-digicatapult) [inline src/AppBar/AppBar.tsx:27]: wrapper should be a `nav` not a `div` as this ia a landmark area. Similarly we should enforce the navigation entries be housed in a `ul` with `li` elements

**Review by mattdean-digicatapult — COMMENTED** [review]
Some strong suggestions to look at fixing before we merge. On the whol getting very close and I love the theming solution

File: src/AppBar/AppBar.tsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+  display: flex;
+  width: ${({ width }) => width || '100%'};
+  flex-direction: row;
+  justify-content: space-between;
+  box-sizing: border-box;
+  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
+  box-shadow: ${({ shadow }) =>
+    shadow
+      ? '0px 2px 4px -1px rgb(0 0 0 / 20%), 0px 4px 5px 0px rgb(0 0 0 / 14%), 0px 1px 10px 0px rgb(0 0 0 / 12%);'
```
[the remark this answers is not in the document]
Comment (n3op2) [reply src/AppBar/AppBar.tsx:36]: copy paste from internet:)

File: src/AppBar/AppBar.tsx (hunk 4 above)
Comment (n3op2) [reply src/AppBar/AppBar.tsx:27]: :+1: originally wanted to do ul/li and make some more general, but then realised that app bar is quite specific already, but you are right, it should be ul/li,, will update

File: src/AppBar/AppBar.tsx (hunk 3 above)
Comment (n3op2) [reply src/AppBar/AppBar.tsx:69]: ok! will convert that as well

File: src/AppBar/AppBar.tsx (hunk 4 above)
Comment (n3op2) [reply src/AppBar/AppBar.tsx:26]: done

File: src/AppBar/AppBar.tsx (hunk 2 above)
Comment (n3op2) [reply src/AppBar/AppBar.tsx]: removed

**Comment by mattdean-digicatapult** [comment]
Not that this is necessary but we're also missing a couple of tests in coverage:
Screenshot 2022-12-14 at 08 29 48

**Review by mattdean-digicatapult — APPROVED** [review]

**Pull request merged**